src/IOutil.f90

```diff
@@ -244,4 +251,30 @@ SUBROUTINE GET_CURRENT_DIRECTORY_internal(dir,length,status)
       IF(status == 0) dir=tmp(1:length-1) !Remove the C_NULL_CHAR
     ENDSUBROUTINE GET_CURRENT_DIRECTORY_internal
 !
+!-------------------------------------------------------------------------------
+    SUBROUTINE MAKE_DIRECTORY_string(dir,status)
+      TYPE(StringType),INTENT(IN) :: dir
+      INTEGER(SIK),INTENT(OUT),OPTIONAL :: status
+      CALL MAKE_DIRECTORY_char(CHAR(dir),status)
+    ENDSUBROUTINE MAKE_DIRECTORY_string
+!
+!-------------------------------------------------------------------------------
+    SUBROUTINE MAKE_DIRECTORY_char(dir,status)
```

Comment:
Note that "status" must be passed into this routine to avoid a segfault. Maybe it should not be optional?